enhancements/proposals/1978-scylladbdatacenter-api/README.md

```diff
@@ -0,0 +1,764 @@
+	// +optional
+	Nodes *int32 `json:"nodes"`
+
+	// topologyLabelSelector specifies a label selector which will be used to target nodes at specified topology constraints.
+	// Datacenter topologyLabelSelector is merged with rack topologyLabelSelector and then converted into nodeAffinity
+	// targeting nodes having specified topology.
+	TopologyLabelSelector *metav1.LabelSelector `json:"topologyLabelSelector,omitempty"`
+
+	// rackTemplate provides a template for every rack.
+	// Every rack inherits properties specified in the template, unless it's overwritten on the rack level.
+	// +optional
+	RackTemplate *RackTemplate `json:"rackTemplate,omitempty"`
+
+	// racks specify the racks in the datacenter.
+	// +optional
+	Racks []RackSpec `json:"racks"`
+
+	// automaticOrphanedNodeCleanup controls if automatic orphan node cleanup should be performed.
+	AutomaticOrphanedNodeCleanup bool `json:"automaticOrphanedNodeCleanup,omitempty"`
+
+	// minTerminationGracePeriodSeconds specifies minimum duration in seconds to wait before every drained node is
+	// terminated. This gives time to potential load balancer in front of a node to notice that node is not ready anymore
+	// and stop forwarding new requests.
+	// This applies only when node is terminated gracefully.
+	// If not provided, Operator will determine this value.
+	// EXPERIMENTAL. Do not rely on any particular behaviour controlled by this field.
+	// +optional
+	MinTerminationGracePeriodSeconds *int32 `json:"minTerminationGracePeriodSeconds,omitempty"`
+
+	// minReadySeconds is the minimum number of seconds for which a newly created ScyllaDB node should be ready
+	// for it to be considered available.
+	// When used to control load balanced traffic, this can give the load balancer in front of a node enough time to
+	// notice that the node is ready and start forwarding traffic in time. Because it all depends on timing, the order
+	// is not guaranteed and, if possible, you should use readinessGates instead.
+	// If not provided, Operator will determine this value.
+	// +optional
+	MinReadySeconds *int32 `json:"minReadySeconds,omitempty"`
+
+	// readinessGates specifies custom readiness gates that will be evaluated for every ScyllaDB Pod readiness.
+	// It's projected into every ScyllaDB Pod as its readinessGate. Refer to upstream documentation to learn more
+	// about readiness gates.
+	// +optional
+	ReadinessGates []corev1.PodReadinessGate `json:"readinessGates,omitempty"`
+}
+
+type ObjectTemplateMetadata struct {
+	// labels is a custom key value map that gets merged with managed object labels.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// annotations is a custom key value map that gets merged with managed object annotations.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+type RackTemplate struct {
+	// nodes specifies desired number of nodes in rack.
+	// +optional
+	Nodes *int32 `json:"nodes"`
```

Comment:
it would be good to describe somewhere how `.nodes`, `rackTemplate.nodes` and `racks[].nodes` merged and which combinations are forbidden